api/src/main/java/org/apache/iceberg/expressions/NamedReference.java

```diff
@@ -38,14 +37,12 @@ public String name() {
 
   @Override
   public BoundReference<T> bind(Types.StructType struct, boolean caseSensitive) {
```

Comment:
Another choice is to add a new API to support binding against schema directly. However, that way needs to modify many places.